### ISSUE TITLE
[java] UnusedLocalVariable has false positive of parameterized array assignment.

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2922]
**Affects PMD Version:** 
6.29.0

**Description:**

False positive for [UnusedLocalVariable](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedlocalvariable) when using parameterized type array assignment.

**Code Sample demonstrating the issue:**

```java
package jls;

import java.util.Objects;

public final class UnsafeVarargs<T> {

  private UnsafeVarargs() {
    throw new IllegalStateException("Private constructor.");
  }

  public static <T> T[] unsafe(T... elements) {
    // unsafe! don't ever return a parameterized varargs array
    return elements;
  }

  public static <T> T[] broken(T seed) {
    // broken! This will be an Object[] no matter what T is
    T[] plant = unsafe(seed, seed, seed);
    return plant;
  }

  public static String[] plant() {
    // ClassCastException
    String[] plants = broken("seed");
    return plants;
  }

  @SuppressWarnings("PMD.SystemPrintln")
  public static void main(String... args) {
    System.out.println(Objects.toString(plant()));
  }
}

```

**Steps to reproduce:**

Run rule UnusedLocalVariable on above code.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by adangel** [comment]
I can't reproduce it. On which line do you get the violation?

**Comment by Fernal73** [comment]
I can't reproduce this either. I'm sorry. I don't know how this could have occurred. Closing...
